### PR TITLE
Add Height, Head Circumference, and BMI

### LINKED ISSUE #191: Feature Request:  Track baby height
Hello again !
I haven't seen a place to track baby height, 
This would be a good feature If this could be added :)
Regards

### PR DESCRIPTION
**Pull request by Daegalus**
Closes #191 

I am an expecting father, and learned these are expected metrics that get checked at every doctors visit on a newborn. We figure we can track some of these ourselves at home.

I found BabyBuddy its amazing, and will be contributing more (planning to tackle Medicine/Immunizations/Shots next (just tracking, no notifications/alerts). But figured I would get my feed wet with this task.

Changes in this PR:

* Added Height, Head Circumference, and BMI (following existing patterns from weight)
* Added Reports for all 3
* Updated Statistics dashboard card with new entries.
* Created a new navigation dropdown `Measurements` and moved `Temperature` and `Weight` into that menu, alongside the new measurements.
* Added links to the reports on the dropdown for reports.
* Added new icons to Fontello and updated files accordingly
* Updated Django localization file for `en_GB`.
* Updated tests to include changes.
* Updated OpenAPI Schema file.
* Generated new migrations for the changes.
* Ran all tests to make sure they pass.

Let me know if I missed anything.

### CODE REVIEW

**Comment by cdubz** [comment]
@Daegalus thanks for contributing! I have been hesitant to add anything more than the "weight" and "temperature" measurements stuff but it looks like you've taken a good approach for it here and done an impressive first pass of work.

I have approved the workflow runs and there are a few linting issues to be resolved there.

I have also approved the Heroku branch deployment so we can see this in action at https://babybuddy-pr-360.herokuapp.com. I ran the `fake` command in that environment and it worked but it didn't add any of the new data. I see that you added the methods to create the data but you need to also add calls to those methods in [`_add_child_data`](https://github.com/babybuddy/babybuddy/blob/e566f76fc5c10d55b568c3a36c04f9ac6cc107c8/babybuddy/management/commands/fake.py#L61) for the data to be populated with each fake child.

For review, I'm traveling right now so not quite sure when I'll be able to really dig in to this. I'll also ping some of the regular contributors/users on Gitter for review/opinions.

Re: medical information -- I'm on board to add that as well but lets keep notifications in mind, at least. See #66, #153, #200.

See also the discussion in #217.

File: openapi-schema.yml [hunk 1]
```diff
@@ -1,8 +1,7 @@
 openapi: 3.0.2
```
Comment (cdubz) [inline openapi-schema.yml:1]: How did you generate this schema? Just curious as it seems to add some stuff that was missing. I have always just set this file to the output of `http://127.0.0.1:8000/api/schema?format=openapi` from a local instance but apparently that is missing some stuff...

File: babybuddy/management/commands/fake.py [hunk 2]
```diff
@@ -249,3 +249,60 @@ def _add_weight_entry(self):
...
+        notes = ''
+        if choice([True, False, False, False]):
+            notes = ' '.join(self.faker.sentences(randint(1, 5)))
+
+        models.bmi.objects.create(
+            child=self.child,
+            bmi=round(self.bmi, 2),
+            date=self.time.date(),
+            notes=notes
+        ).save()
```
Comment (cdubz) [inline babybuddy/management/commands/fake.py:308]: Per my [pre-review comment](https://github.com/babybuddy/babybuddy/pull/360#issuecomment-1002599632) -- please add calls to these new methods to [`_add_child_data`](https://github.com/babybuddy/babybuddy/blob/e566f76fc5c10d55b568c3a36c04f9ac6cc107c8/babybuddy/management/commands/fake.py#L61).

**Review by cdubz — CHANGES_REQUESTED** [review]
Well a big PR but easy enough to review when its so well done. Thanks!

Just two review comments to look at and the linting issue and this should be good.

File: openapi-schema.yml (hunk 1 above)
Comment (Daegalus) [reply openapi-schema.yml:1]: From the official Django Docs on RestFramework: https://www.django-rest-framework.org/api-guide/schemas/#generating-a-static-schema-with-the-generateschema-management-command

`python manage.py generateschema --file openapi-schema.yml`

File: babybuddy/management/commands/fake.py (hunk 2 above)
Comment (Daegalus) [reply babybuddy/management/commands/fake.py:308]: on it, just got home from a day of errands so going to tackle this now

File: openapi-schema.yml (hunk 1 above)
Comment (cdubz) [reply openapi-schema.yml:1]: Cool. It’s interesting that the generated one from the API appears to be missing stuff. I’ll need to add a gulp command for that 👍

**Comment by Daegalus** [comment]
> @Daegalus thanks for contributing! I have been hesitant to add anything more than the "weight" and "temperature" measurements stuff but it looks like you've taken a good approach for it here and done an impressive first pass of work.
> 
> I have approved the workflow runs and there are a few linting issues to be resolved there.
> 
> I have also approved the Heroku branch deployment so we can see this in action at https://babybuddy-pr-360.herokuapp.com. I ran the `fake` command in that environment and it worked but it didn't add any of the new data. I see that you added the methods to create the data but you need to also add calls to those methods in [`_add_child_data`](https://github.com/babybuddy/babybuddy/blob/e566f76fc5c10d55b568c3a36c04f9ac6cc107c8/babybuddy/management/commands/fake.py#L61) for the data to be populated with each fake child.
> 
> For review, I'm traveling right now so not quite sure when I'll be able to really dig in to this. I'll also ping some of the regular contributors/users on Gitter for review/opinions.
> 
> Re: medical information -- I'm on board to add that as well but lets keep notifications in mind, at least. See #66, #153, #200.
> 
> See also the discussion in #217.

NP, I will take care of these now. And I have some ideas on how we can do Notifications now with the current structure, while you guys work on the new generic model design. That way we can get some of these features added before 2.0's rewrite.

I would help but I am not familiar with Django, and even if I am a backend/devops engineer, its not something I work in regularly so not familiar with what the idiomatic ways of doing object models is for Python/Django. I live very deep in Go land.

**Comment by cdubz** [comment]
Oops — #361 just added a migration. You’ll need to update yours to follow that one.

**Comment by Daegalus** [comment]
> Oops — #361 just added a migration. You’ll need to update yours to follow that one.

should be fixed

**Comment by Daegalus** [comment]
also noticed the font doesn't seem to be updating on the Heroku deploy, as the icons for the new menu options aren't there. Might need a hard refresh?

**Review by cdubz — APPROVED** [review]

**Comment by cdubz** [comment]
Looks great! Re: the icons -- when adding new ones it still necessary to run `gulp updatestatic` as well to pull in the CSS for the new icons.

Thanks, @Daegalus!

**Pull request merged**